### ISSUE TITLE
SMTP connect() failed

### ISSUE THREAD

**Issue by ghost** [nilsteampassnet/TeamPass#1822]
I'm trying to configure email, but this one giving SMTP connect () failed error, I'm using company domain, has some configuration that needs to be done in phpmailer.
The administrator account is with email configured.

![sem titulo](https://user-images.githubusercontent.com/30321447/28466659-7ffa0404-6e03-11e7-8408-977c07895458.png)

### Steps to reproduce
1.
2.
3.

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Server configuration
**Operating system**:

**Web server:**

**Database:*Mysql

**PHP version:5.6

**Teampass version:2.1.27.7

**Updated from an older Teampass or fresh install:**

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### Firebug log ([How to?](http://teampass.net/2014-02-09-how-to-communicate-an-error-log))
```
Insert the Firebug log here
```

**Comment by nilsteampassnet** [comment]
I am surprised to see that the SMTP port of your server is 587 when it is not securized.
It is often port 25 that is used in such case.

Please check the email server configuration.

**Comment by rafalin** [comment]
I have reviewed the settings but still giving error when sending. It needs to be done in php configurations.

![sem titulo](https://user-images.githubusercontent.com/30321447/28537235-803b115e-7080-11e7-9eb4-7a7af0fbb4ec.png)

**Comment by nilsteampassnet** [comment]
Are you sure of your port number?

**Comment by rafalin** [comment]
Yes, this is the port used for smtp.

**Comment by Florian7** [comment]
I have exactly the same problem with a basic smtp server which work anywhere else with this same config:
<img width="666" alt="capture d ecran 2017-07-29 a 21 39 27" src="https://user-images.githubusercontent.com/30557758/28747764-eb3a9bbc-74a6-11e7-84ef-471f575c852b.png">

I have tested on php 5.6.30, Teampass 2.1.27.7, apache 2.4.10 and Debian 8.9 
and php 7.0.19, Teampass 2.1.27.7, apache 2.4.25 and Debian 9.1

I have tested with an authentification to, but it sould work without.

**Comment by nilsteampassnet** [comment]
Does your admin user have an email defined?

**Comment by Florian7** [comment]
Yes, the email is defined. Actually, when it is not, the error does not contain "SMTP connect() failed. ..."

**Comment by burnbabyburn** [comment]
Actually, maybe your smtp server **is** blocking your teampass server with something like an ip-based acl whitelist? Especially in a company enviroment this wouldn't be uncommon ;)

**Comment by Florian7** [comment]
@burnbabyburn thanks for the advice with sendmail, I'll do some tests to check it is well configured.
But something even stranger, I tried to install Teampass on a Raspberry Pi with raspbian jessie lite and php 5.6.30, and there, no problem to send the mail... I checked that sendmail was configured, but it is absolutely not installed... Maybe php can use an other service ?

**Comment by nilsteampassnet** [comment]
The library PHPMailer used in Teampass doesn't require an SMTP server to work locally or PHP sendMail function. It can also use Mail function which is certainly the case with the Raspberry.

SO here I think it more the configuration that requires to be verified.

**Comment by Florian7** [comment]
Damn it!! ... I finally got the answer !!

I tried different url linked to my mail server, and finally found one which work !!
My tried:
-Local IP address:     172.16.1.11
-Published url:           mail.******.ch (which works on my raspberry, out of the local network)
-Local url:                  srvexch.*****.ch (record on our local DNS)

This last one works. So I can send mail now.

But I also checked the smtp config on an other platform (GLPI, for ticketing). Then, on an old installation, the smtp works perfectly with all three configs exposed over there.

<img width="906" alt="old_glpi" src="https://user-images.githubusercontent.com/30557758/28896507-01ce2f2c-77dd-11e7-8153-8b0e241b08c6.png">


But, it is on an old web server in parallel (the one were is installed teampass is a new one which is supposed to replace it). So it installed the all new version of GLPI on this new server and, surprise, it didn't worked except with the local url, like for Teampass... but where is the problem ? The server ? The firewall rules ?

<img width="907" alt="new_glpi_doesn t_work" src="https://user-images.githubusercontent.com/30557758/28896553-2da63a2c-77dd-11e7-870e-4ab42ed816cc.png">


Actually not. It is only this little "checkbox" which permitted me to find the answer. 

<img width="898" alt="new_glpi_works" src="https://user-images.githubusercontent.com/30557758/28896570-3c4679d4-77dd-11e7-93fc-6858f7060f5b.png">


So, my conclusion is that I have a problem with the certificates of the mail server. The IP address doesn't have one, of course, and mail.******.ch seems to be certified only for external connections. Only srvexch.******.ch is certified in intern.

Teampass seems to verify automatically the certificate, so if you don't have one, you'll get some issues.

Maybe @nilsteampassnet could confirm that, and eventually see if there is a solution to pass over this verification.
